File: helfomat-api/src/main/java/de/helfenkannjeder/helfomat/api/organization/event/OrganizationEditDeleteContactPersonEventDto.java
package de.helfenkannjeder.helfomat.api.organization.event;

import de.helfenkannjeder.helfomat.api.organization.ContactPersonDto;
import de.helfenkannjeder.helfomat.core.organization.OrganizationId;

/**
 * @author Valentin Zickner
 */
public class OrganizationEditDeleteContactPersonEventDto extends OrganizationEventDto {
    private ContactPersonDto contactPerson;

    OrganizationEditDeleteContactPersonEventDto() {
    }

    public OrganizationEditDeleteContactPersonEventDto(OrganizationId organizationId, ContactPersonDto contactPerson) {
        super(organizationId);
        this.contactPerson = contactPerson;
    }

    public ContactPersonDto getContactPerson() {
        return contactPerson;
    }

    @Override
    public <T> T visit(OrganizationEventDtoVisitor<T> visitor) {
        return visitor.visit(this);
    }

}

File: helfomat-api/src/main/kotlin/de/helfenkannjeder/helfomat/api/organization/event/OrganizationEditDeleteContactPersonEventDto.kt
package de.helfenkannjeder.helfomat.api.organization.event

import de.helfenkannjeder.helfomat.api.organization.ContactPersonDto
import de.helfenkannjeder.helfomat.core.organization.OrganizationId

/**
 * @author Valentin Zickner
 */
data class OrganizationEditDeleteContactPersonEventDto(
    override val organizationId: OrganizationId,
    val contactPerson: ContactPersonDto,
    override val eventApplicable: Boolean
) : OrganizationEventDto {

    override fun <T> visit(visitor: OrganizationEventDtoVisitor<T>): T {
        return visitor.visit(this)
    }

}